### PR TITLE
Inject Hazelcast instance into quorum function

### LINKED ISSUE #7499: Support HazelcastInstanceAware for QuorumFunction
If a `QuorumFunction` implements `HazelcastInstanceAware`, it should have instance injected.

### PR DESCRIPTION
**Pull request by jerrinot**
Fixes #7499

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/quorum/QuorumTest.java [hunk 1]
```diff
@@ -227,6 +229,49 @@ public boolean apply(Collection<Member> members) {
         }
     }
 
+    @Test
+    public void givenQuorumFunctionConfigured_whenImplementsHazelcastInstanceAware_thenHazelcastInjectsItsInstance() throws Exception {
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(1);
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/quorum/QuorumTest.java]: A HZ instance factory in a local variable looks suspicious to me. Are those instances shutdown properly?

File: hazelcast/src/test/java/com/hazelcast/quorum/QuorumTest.java [hunk 2]
```diff
@@ -227,6 +229,49 @@ public boolean apply(Collection<Member> members) {
...
+            this.instance = instance;
+        }
+
+        @Override
+        public boolean apply(Collection<Member> members) {
+            return false;
+        }
+    }
+
+
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/quorum/QuorumTest.java]: If you have to touch this class, this superfluous newline could also be removed :)

File: hazelcast/src/test/java/com/hazelcast/quorum/QuorumTest.java [hunk 3]
```diff
@@ -227,6 +229,49 @@ public boolean apply(Collection<Member> members) {
         }
     }
 
+    @Test
+    public void givenQuorumFunctionConfigured_whenImplementsHazelcastInstanceAware_thenHazelcastInjectsItsInstance() throws Exception {
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/quorum/QuorumTest.java]: I can't see it on GitHub, but if the tests don't need it you could remove the `throws Exception` to shorten the lines a bit (the test names are already quite long).

File: hazelcast/src/main/java/com/hazelcast/quorum/impl/QuorumImpl.java [hunk 4]
```diff
@@ -200,6 +202,9 @@ private void initializeQuorumFunction() {
                 throw ExceptionUtil.rethrow(e);
             }
         }
+        if (quorumFunction instanceof HazelcastInstanceAware) {
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/quorum/impl/QuorumImpl.java]: maybe doing this check in the end of the method makes it less fragile

**Pull request merged**